### PR TITLE
Fix ordering of ls -t, which was backwards

### PR DESCRIPTION
**Pull request by ids1024**

### CODE REVIEW

**Comment by ebfe** [comment]
Now -S is wrong?

**Comment by ids1024** [comment]
Oops; yes, I hadn't noticed that this code handles multiple kinds of sort.

**Comment by ids1024** [comment]
It should be correct now.

File: src/ls/ls.rs [hunk 1]
```diff
@@ -196,6 +196,7 @@ fn sort_entries(entries: &mut Vec<PathBuf>, options: &getopts::Matches) {
                     .unwrap_or(std::time::UNIX_EPOCH)
             });
         }
+        entries.reverse(); // Newest first
```
Comment (Arcterus) [inline src/ls/ls.rs]: Perhaps it makes more sense to change `sort_by_key` above into:
```rust
entries.sort_by(|a, b| {
    get_mod_time(b, options).cmp(get_mod_time(a, options)))
});
```
where `get_mod_time()` is defined as:
```rust
fn get_mod_time(path: &PathBuf, options: &getopts::Matches) -> time::SystemTime {
    get_metadata(k, options)
        .and_then(|md| md.modified())
        .unwrap_or(std::time::UNIX_EPOCH)
}
```

That way we don't have to reverse the vector.  Of course, you would also have to define a similar `get_create_time` method or something to that effect for Unix.

Comment (ids1024) [inline src/ls/ls.rs]: Good point; it seems the standard library has a neat struct for reversing sorts, so I used that.

File: src/ls/ls.rs [hunk 2]
```diff
@@ -191,9 +192,10 @@ fn sort_entries(entries: &mut Vec<PathBuf>, options: &getopts::Matches) {
             entries.sort_by_key(|k| get_metadata(k, options).map(|md| md.ctime()).unwrap_or(0));
         } else {
             entries.sort_by_key(|k| {
-                get_metadata(k, options)
+                // Newest first
+                Reverse(get_metadata(k, options)
```
Comment (Arcterus) [inline src/ls/ls.rs]: I didn't even know this was a thing lmao

File: src/ls/ls.rs [hunk 3]
```diff
@@ -191,9 +192,10 @@ fn sort_entries(entries: &mut Vec<PathBuf>, options: &getopts::Matches) {
             entries.sort_by_key(|k| get_metadata(k, options).map(|md| md.ctime()).unwrap_or(0));
```
Comment (Arcterus) [inline src/ls/ls.rs]: Oh wait, I may have merged this too quickly.  Does this need to be reversed as well?

Comment (ids1024) [inline src/ls/ls.rs]: Oh yes, I suppose it does.

**Pull request merged**